src/plugins/dataflow/model/dataflow-content.ts

```diff
@@ -105,10 +107,16 @@ export const DataflowContentModel = TileContentModel
         `    "dy": ${zoom.dy},`,
         `    "scale": ${zoom.scale}`,
         `  },`,
+        // `  "programRecordingMode: ${self.programRecordingMode}"`,
         `  "program": ${stringify(self.programWithoutRecentValues())}`,
         `}`
       ].join("\n");
     },
+    // ------ADDED---------------------------------
+    get isEmptyDataSet(){
```

Comment:
@dennisrcao Based on the conversation with Kirk, I think this should be named:
`isDataSetEmptyCases`. That differentiates it from basic `isEmpty`.